tests/conftest.py

```diff
@@ -713,3 +713,77 @@ def rook_repo(request):
     get_rook_repo(
         config.RUN['rook_branch'], config.RUN.get('rook_to_checkout')
     )
+
+
+@pytest.fixture(scope="session")
+def memory_leak_function(request):
```

Comment:
great stuff!

Reply:
Yes Vasu it will be at test level, will add more details in the function about usage